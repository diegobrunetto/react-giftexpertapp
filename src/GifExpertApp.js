import React, { useState } from "react";

import "nes.css/css/nes.min.css";

import { AddCategory } from "./components/AddCategory";
import { GifGrid } from "./components/GifGrid";

import "./index.css";

const GifExpertApp = ({ defaultCategories = [] }) => {
  const [categories, setCategories] = useState(defaultCategories);

  return (
    <>
      <h2 className="nes-text">Busca tus Gifs favoritos</h2>
      <AddCategory setCategories={setCategories} />
      <ul>
        {categories.map((category) => (
          <GifGrid category={category} key={category} />
        ))}
      </ul>
    </>
  );
};

export default GifExpertApp;
