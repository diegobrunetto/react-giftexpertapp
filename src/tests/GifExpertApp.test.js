import React from "react";
import "@testing-library/jest-dom";

import { shallow } from "enzyme";
import GifExpertApp from "../GifExpertApp";

describe("Pruebas en GifExpertApp", () => {
  test("debe mostrar correctamente el componente", () => {
    const wrapper = shallow(<GifExpertApp />);

    expect(wrapper).toMatchSnapshot();
  });

  test("debe mostrat la lista de categorias", () => {
    const categories = ["Kobe", "Jordan", "Lebron"];
    const wrapper = shallow(<GifExpertApp defaultCategories={categories} />);

    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find("GifGrid").length).toBe(categories.length);
  });
});
  