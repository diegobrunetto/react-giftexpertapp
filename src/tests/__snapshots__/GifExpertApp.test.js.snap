// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en GifExpertApp debe mostrar correctamente el componente 1`] = `
<Fragment>
  <h2
    className="nes-text"
  >
    Busca tus Gifs favoritos
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <ul />
</Fragment>
`;

exports[`Pruebas en GifExpertApp debe mostrat la lista de categorias 1`] = `
<Fragment>
  <h2
    className="nes-text"
  >
    Busca tus Gifs favoritos
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <ul>
    <GifGrid
      category="Kobe"
      key="Kobe"
    />
    <GifGrid
      category="Jordan"
      key="Jordan"
    />
    <GifGrid
      category="Lebron"
      key="Lebron"
    />
  </ul>
</Fragment>
`;
